test-suite/src/test/scala/org/scalajs/testsuite/javalib/CollectionTest.scala

```diff
@@ -93,7 +91,11 @@ trait CollectionTest[F <: CollectionFactory] extends JasmineTest {
     }
 
     it("should store custom objects") {
-      case class TestObj(num: Int)
+      case class TestObj(num: Int) extends jl.Comparable[TestObj] {
+        def compareTo(o: TestObj): Int =
+          if (o == null) -1
+          else num.compareTo(o.num)
```

Comment:
The comment was about why extending with `js.Comparable` at all. It was because some collections use the natural comparator. 

Yes, I agree that the `o == null` might be to overcome a bug in one of the collections. I'll check the source of the exception that made me add this case.